### PR TITLE
updates to nvcc_wrapper to match the 4.0RC default arch and look for c source files

### PR DESCRIPTION
**Pull request by ARSanderson**
Add *.c source files to be compiled as cuda files by `nvcc_wrapper`

### CODE REVIEW

**Review by PhilMiller — APPROVED** [review]

**Comment by PhilMiller** [comment]
ok to test

**Comment by PhilMiller** [comment]
Kinda connected to #5914

File: bin/nvcc_wrapper [hunk 1]
```diff
@@ -12,8 +12,8 @@
 # or g++ as their back-end compiler.  The defaults can be overwritten
 # by using the usual arguments (e.g., -arch=sm_30 -ccbin icpc).
 
-default_arch="sm_35"
-#default_arch="sm_50"
+default_arch="sm_70"
+#default_arch="sm_80"
```
Comment (dalg24) [inline bin/nvcc_wrapper:16]: Your PR is based on a version of from Aug 20, 2020
[5f931d4](https://github.com/kokkos/kokkos/commit/5f931d450026abbdb283ad8f30764601202db35a)

This change is already in the current `develop`.
https://github.com/kokkos/kokkos/blob/c5fe10e7f9e4925f0269f1eb155770935117f2d4/bin/nvcc_wrapper#L17-L18

Please fetch the latest version and re-apply your patch on top of it.
I can assist if needed.

Comment (ARSanderson) [reply bin/nvcc_wrapper:16]: Sorry about that ... I did a fetch to update the fork and thought it updated. Let me go back and see what happened.

**Comment by PhilMiller** [comment]
Please just rebase that one commit, then. Or, I'll see if I can just do that and push it.

@dalg24 Sorry I didn't notice the ancient history issue

**Comment by PhilMiller** [comment]
I just pushed the rebase

**Comment by PhilMiller** [comment]
And re-edited the commit message to match the current contents.

**Comment by PhilMiller** [comment]
Ubuntu Clang failure is just CI noise, and can be ignored here.

**Comment by PhilMiller** [comment]
The HIP failure on Jenkins is the usual `hipOutOfMemory` junk, and can also be ignored.

**Comment by ARSanderson** [comment]
I am closing this PR as the change is not needed. See https://github.com/kokkos/kokkos/issues/5914

@PhilMiller and @dalg24 My profuse apologizes for the confusion and the needless work I thought I was helping by getting that change in.

**Comment by dalg24** [comment]
No worries :)

**Pull request closed without merging**